src/overlays/effects/ovl_Effect_Ss_D_Fire/z_eff_ss_d_fire.c

```diff
@@ -1,31 +1,114 @@
 /*
  * File: z_eff_ss_d_fire.c
  * Overlay: ovl_Effect_Ss_D_Fire
- * Description:
+ * Description: Dodongo Fire
  */
 
 #include "z_eff_ss_d_fire.h"
+#include "objects/object_dodongo/object_dodongo.h"
+
+#define rScale regs[0]
+#define rTexIndex regs[1]
+#define rAlpha regs[5]
+#define rAlphaStep regs[6]
+#define rFadeDelay regs[7]
+#define rScaleStep regs[8]
+#define rObjectIndex regs[9]
 
 #define PARAMS ((EffectSsDFireInitParams*)initParamsx)
 
-s32 EffectSsDFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsDFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsDFire_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsDFire_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_D_Fire_InitVars = {
     EFFECT_SS_D_FIRE,
     EffectSsDFire_Init,
 };
 
-#endif
+static TexturePtr sFireTextures[] = { gDodongoFire0Tex, gDodongoFire1Tex, gDodongoFire2Tex, gDodongoFire3Tex };
+
+s32 EffectSsDFire_CheckForObject(EffectSs* this, PlayState* play) {
+    if (((this->rObjectIndex = Object_GetIndex(&play->objectCtx, OBJECT_DODONGO)) < 0) ||
```

Comment:
```suggestion
s32 EffectSsDFire_CheckForObject(EffectSs* this, PlayState* play) {
    this->rObjectIndex = Object_GetIndex(&play->objectCtx, OBJECT_DODONGO)
    if ((this->rObjectIndex < 0) ||
```
Does it need to be one line?

Reply:
Unfortunately yes. It is actually a pretty common pattern in effects.